libparsec/crates/platform_device_loader/src/native/mod.rs

```diff
@@ -34,38 +34,47 @@ impl From<keyring::Error> for SaveDeviceError {
  * List available devices
  */
 
-fn find_device_files(path: PathBuf) -> Vec<PathBuf> {
+fn find_device_files(path: &Path) -> Result<Vec<PathBuf>, ListAvailableDeviceError> {
     // TODO: make file access on a worker thread !
 
-    let mut key_file_paths = vec![];
-
     // `fs::read_dir` fails if path doesn't exists, is not a folder or is not
     // accessible... In any case, there is not much we can do but to ignore it.
-    if let Ok(children) = std::fs::read_dir(path) {
-        for path in children.filter_map(|entry| entry.as_ref().map(std::fs::DirEntry::path).ok()) {
-            if path.extension() == Some(DEVICE_FILE_EXT.as_ref()) {
-                key_file_paths.push(path)
-            } else if path.is_dir() {
-                key_file_paths.append(&mut find_device_files(path))
+    match std::fs::read_dir(path) {
```

Comment:
Comment above should be updated, the error is no longer ignored.

Reply:
Done